main/main.cpp

```diff
@@ -2402,133 +2402,151 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
-		if (rendering_method != "forward_plus" &&
-				rendering_method != "mobile" &&
-				rendering_method != "gl_compatibility" &&
-				rendering_method != "dummy") {
-			OS::get_singleton()->print("Unknown rendering method '%s', aborting.\nValid options are ",
-					rendering_method.utf8().get_data());
-
-			Vector<String> rendering_method_hints = renderer_hints.split(",");
-			rendering_method_hints.push_back("dummy");
-			for (int i = 0; i < rendering_method_hints.size(); i++) {
-				if (i == rendering_method_hints.size() - 1) {
-					OS::get_singleton()->print(" and ");
-				} else if (i != 0) {
-					OS::get_singleton()->print(", ");
-				}
-				OS::get_singleton()->print("'%s'", rendering_method_hints[i].utf8().get_data());
-			}
-
-			OS::get_singleton()->print(".\n");
-			goto error;
+		if (!rendering_method_hints.is_empty()) {
+			renderer_hints = String(",").join(rendering_method_hints);
+		} else {
+			renderer_hints = "dummy";
 		}
-	}
-	if (renderer_hints.is_empty()) {
-		renderer_hints = "dummy";
-	}
-
-	if (!rendering_driver.is_empty()) {
-		// As the rendering drivers available may depend on the display driver and renderer
-		// selected, we can't do an exhaustive check here, but we can look through all
-		// the options in all the display drivers for a match.
+		rendering_method_hints.append("dummy");
 
-		bool found = false;
+		// Deduplicate driver entries, as a rendering driver may be supported by several display servers.
+		Vector<String> unique_rendering_drivers;
 		for (int i = 0; i < DisplayServer::get_create_function_count(); i++) {
 			Vector<String> r_drivers = DisplayServer::get_create_function_rendering_drivers(i);
 
 			for (int d = 0; d < r_drivers.size(); d++) {
-				if (rendering_driver == r_drivers[d]) {
-					found = true;
-					break;
+				if (!unique_rendering_drivers.has(r_drivers[d])) {
+					unique_rendering_drivers.append(r_drivers[d]);
 				}
 			}
 		}
 
-		if (!found) {
-			OS::get_singleton()->print("Unknown rendering driver '%s', aborting.\nValid options are ",
-					rendering_driver.utf8().get_data());
-
-			// Deduplicate driver entries, as a rendering driver may be supported by several display servers.
-			Vector<String> unique_rendering_drivers;
-			for (int i = 0; i < DisplayServer::get_create_function_count(); i++) {
-				Vector<String> r_drivers = DisplayServer::get_create_function_rendering_drivers(i);
+#ifdef GLES3_ENABLED
+		Vector<String> gl_rendering_drivers;
+#endif
+#if RD_ENABLED
+		Vector<String> rd_rendering_drivers;
+#endif
 
-				for (int d = 0; d < r_drivers.size(); d++) {
-					if (!unique_rendering_drivers.has(r_drivers[d])) {
-						unique_rendering_drivers.append(r_drivers[d]);
-					}
+		for (int i = 0; i < unique_rendering_drivers.size(); i++) {
+#if GLES3_ENABLED
+			// Use begins_with to directly match all OpenGL variants.
+			if (unique_rendering_drivers[i].begins_with("opengl")) {
+				gl_rendering_drivers.append(unique_rendering_drivers[i]);
+			} else {
+#endif
```

Comment:
```suggestion
			} else
#endif
			{
```
Or you'll have an unmatched bracket